src/event_server.cpp

```diff
@@ -2087,6 +2087,47 @@ static void get_calibration_data(JObj& response, const json_value *params)
     response << jrpc_result(rslt);
 }
 
+static void set_cooler_state(JObj& response, const json_value *params)
+{
+    Params p("enabled", params);
+    const json_value *val = p.param("enabled");
+    bool enable;
+    if (!val || !bool_param(val, &enable))
+    {
+        response << jrpc_error(JSONRPC_INVALID_PARAMS, "expected enabled boolean param");
+        return;
+    }
+
+    if (!pCamera || !pCamera->Connected)
+    {
+        response << jrpc_error(1, "camera not connected");
+        return;
+    }
+
+    if (!pCamera->HasCooler)
+    {
+        response << jrpc_error(1, "camera lacks a cooler");
+        return;
+    }
+
+    if (!pCamera->SetCoolerOn(enable))
+    {
+        response << jrpc_error(1, "failed to set cooler state");
+    }
+
+    if (enable)
+    {
+        double setpt = pConfig->Profile.GetDouble("/camera/CoolerSetpt", 10.0);
+        if (!pCamera->SetCoolerSetpoint(setpt));

```

Comment:
ditto here (SetCoolerSetpoint returns false on success 🤢 )

Reply:
I agree. As annoying as the current convention is, I've gotten used to it and it is at least consistent.  I think a partial, mixed convention would make life harder and contribute to programming errors.